tslib/readers/pi_xml_reader.py

```diff
@@ -163,121 +186,134 @@ def bulk_get_series(self, chunk_size=250000):
-                    comment = series[-1]
-                    if comment.text is not None:
-                        meta_data[-1][u'comment'] = unicode(comment.text)
+                    # Construct a pandas DataFrame from the events.
+                    dataframe = dataframe_from_bulk(bulk_data, tz_offset)
 
-                series.clear()
-                yield pd.DataFrame(meta_data), dataframe
+                    yield pd.DataFrame(meta_data), dataframe
 
-                # Return empty metadata if all data was yielded or else the
-                # last metadata.
-                meta_data = [] if not bulk_data["value"] else meta_data[-1:]
+                    # keep the last metadata entry
+                    meta_data = meta_data[-1:]
 
-        if bulk_data["value"]:
+        if i > 0:
             # There is still some data left smaller than the chunk size. We
             # don't need to take care of cleaning up.
-            dataframe = take_dataframe_from_bulk(bulk_data, chunk_size)
+
+            for key, value in bulk_data.items():
+                bulk_data[key] = value[:i]
+
+            dataframe = dataframe_from_bulk(bulk_data, tz_offset)
             yield pd.DataFrame(meta_data), dataframe
 
 
-def take_dataframe_from_bulk(bulk_data, chunk_size):
+def dataframe_from_bulk(data, tz_offset):
     """
-    Create a Timeseries dataframe with a maximum of chunk_size rows.
+    Create a Timeseries dataframe from a dict of ndarrays.
 
     Args:
         bulk_data(dict):
-        chunk_size(int): max number of rows for the
+        tz_offset: pandas Offset or None
 
     Returns:
-        data_frame
-
+        pandas DataFrame object
     """
-    data = {}
-
-    def set_data(**columns_and_dtypes):
-        for column, dtype in columns_and_dtypes.items():
-            data[column] = np.array(
-                bulk_data[column][:chunk_size], dtype=dtype)
-            bulk_data[column] = bulk_data[column][chunk_size:]
-
-    def set_if_any(**columns_and_dtypes):
-        set_data(**{
-            column: dtype for column, dtype in columns_and_dtypes.items()
-            if any(bulk_data[column])
-        })
-        for column in columns_and_dtypes.keys():
-            if not any(bulk_data[column]):
-                bulk_data[column] = bulk_data[column][chunk_size:]
-
-    set_data(value=np.float, location_code=str, code=str)
-    set_if_any(flag=np.int32, flag_source=str, comment=str, user=str)
-    data["timestamp"] = bulk_data["timestamp"][:chunk_size]
-    dataframe = pd.DataFrame(data=data)
-    dataframe.set_index(['timestamp', 'location_code', 'code'],
-                        inplace=True, drop=False)
-    bulk_data["timestamp"] = bulk_data["timestamp"][chunk_size:]
+    timestamp = pd.DatetimeIndex(data['timestamp'])
+    if tz_offset is not None:
+        timestamp = timestamp.tz_localize(tz_offset)
+
+    index = pd.MultiIndex.from_arrays(
+        arrays=[
+            pd.Categorical(data['code']),
```

Comment:
Nu werkt deze code niet meer: /code/src/hydra-core/hydra_core/utils/convert.pyc in to_UTC(dataframe, level)
```
/code/src/hydra-core/hydra_core/utils/convert.pyc in to_UTC(dataframe, level)
     76         tz_check = dataframe.index.tz is None
     77     else:
---> 78         tz_check = dataframe.index.levels[level].tz is None
     79     if tz_check:
     80         logger.warning("Time zone is lacking, assuming UTC.")

AttributeError: 'CategoricalIndex' object has no attribute 'tz'
```